### PR TITLE
Merge of branch ngrx-layers

### PR DESCRIPTION
**Pull request by MarkShleifer**
The branch implements the connections of the layers manager to the store and has some examples working with OSM

### CODE REVIEW

File: src/app/effects/map.app.effects.spec.ts [hunk 1]
```diff
@@ -1,3 +1,6 @@
+import { UnselectLayerAction } from './../packages/menu-items/layers-manager/actions/layers.actions';
```
Comment (eldadcohen1) [inline src/app/effects/map.app.effects.spec.ts]: Duplication of layer.actions

File: src/app/effects/map.app.effects.spec.ts [hunk 2]
```diff
@@ -87,4 +90,39 @@ describe('MapAppEffects', () => {
 		});
 	})
 
+	it('addVectorLayer$ should add the selected Layer to the map', () => {
+		let staticLeaf: ILayerTreeNodeLeaf = { name: 'staticLayer', id: 'staticLayerId', isChecked: false, url: "fake_url", isIndeterminate: false, children: <ILayerTreeNode[]>[] };
```
Comment (eldadcohen1) [inline src/app/effects/map.app.effects.spec.ts]: Please cut the line.
Do not write long lines.

File: src/app/effects/map.app.effects.spec.ts [hunk 3]
```diff
@@ -87,4 +90,39 @@ describe('MapAppEffects', () => {
...
+		spyOn(imageryCommunicatorService, 'provideCommunicator').and.callFake(() => imagery1);
+		spyOn(imagery1, 'addVectorLayer');
+
+		mapAppEffects.selectCase$.subscribe(() => {
+			expect(imagery1.addVectorLayer).toHaveBeenCalledWith(staticLeaf);
+		});
+	});
+
+		it('removeVectorLayer$ should remove the unselected Layer to the map', () => {
+		let staticLeaf: ILayerTreeNodeLeaf = { name: 'staticLayer', id: 'staticLayerId', isChecked: false, url: "fake_url", isIndeterminate: false, children: <ILayerTreeNode[]>[] };
```
Comment (eldadcohen1) [inline src/app/effects/map.app.effects.spec.ts]: Same here about lines

File: src/app/packages/imageryOpenLayerMap/map/openLayerMap.ts [hunk 4]
```diff
@@ -63,15 +63,35 @@ export class OpenLayerMap implements IMap {
 		this._mapObject.addLayer(layer);
 	}
 
+	public addVectorLayer(layer: any): void {
```
Comment (eldadcohen1) [inline src/app/packages/imageryOpenLayerMap/map/openLayerMap.ts]: Please add comment to change this to use  "@ansyn/map-source-provider" pkg.

File: src/app/packages/menu-items/layers-manager/actions/layers.actions.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import { ILayerTreeNodeRoot } from './../models/layer-tree-node-root';
```
Comment (eldadcohen1) [inline src/app/packages/menu-items/layers-manager/actions/layers.actions.ts]: Good actions 😄

File: src/app/packages/menu-items/layers-manager/reducers/layers.reducer.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+            return state;
+
+        case LayersActionTypes.LAYER_TREE_LOADED:
+            return Object.assign({}, state, {
+                layers: action.payload.layers,
+                selectedLayers: action.payload.selectedLayers
+            });
+
+        case LayersActionTypes.SELECT_LAYER:
+            let selectedLayerIndex: number = state.selectedLayers.indexOf(action.payload);
```
Comment (eldadcohen1) [inline src/app/packages/menu-items/layers-manager/reducers/layers.reducer.ts]: Check if there is set data structure that handel duplications.

File: src/app/packages/menu-items/layers-manager/services/data-layers.service.spec.ts [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+                }
+            ]
+        };
+
+        spyOn(http, 'get').and.returnValue(Observable.of(new Response(serverResponse)));
+        dataLayersService.getAllLayersInATree('caseId');
+        expect(http.get).toHaveBeenCalledWith(`${dataLayersService.baseUrl}/caseId/layers`);
+    });
+
+    // it('extractData should parse the server data correctly', () => {
```
Comment (eldadcohen1) [inline src/app/packages/menu-items/layers-manager/services/data-layers.service.spec.ts]: Delete unnecessary comments (marked code)

File: src/app/packages/menu-items/layers-manager/services/data-layers.service.ts [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/catch';
+import 'rxjs/add/operator/map';
+
+export type LayerRootsBundle = { layers: ILayerTreeNodeRoot[], selectedLayers: ILayerTreeNodeLeaf[] };
+type LayerNodesBundle = { layers: ILayerTreeNode[], selectedLayers: ILayerTreeNodeLeaf[] };
+
+@Injectable()
+export class DataLayersService {
+  baseUrl = 'http://localhost:9001/api/v1/cases';
```
Comment (eldadcohen1) [inline src/app/packages/menu-items/layers-manager/services/data-layers.service.ts]: Should be in config. Please add comment

File: src/app/effects/map.app.effects.spec.ts (hunk 1 above)
Comment (MarkShleifer) [inline src/app/effects/map.app.effects.spec.ts]: done

File: src/app/effects/map.app.effects.spec.ts (hunk 2 above)
Comment (MarkShleifer) [inline src/app/effects/map.app.effects.spec.ts]: done

File: src/app/effects/map.app.effects.spec.ts (hunk 3 above)
Comment (MarkShleifer) [inline src/app/effects/map.app.effects.spec.ts]: done

File: src/app/packages/imageryOpenLayerMap/map/openLayerMap.ts (hunk 4 above)
Comment (MarkShleifer) [inline src/app/packages/imageryOpenLayerMap/map/openLayerMap.ts]: done

File: src/app/packages/menu-items/layers-manager/actions/layers.actions.ts (hunk 5 above)
Comment (MarkShleifer) [inline src/app/packages/menu-items/layers-manager/actions/layers.actions.ts]: 👌

File: src/app/packages/menu-items/layers-manager/reducers/layers.reducer.ts (hunk 6 above)
Comment (MarkShleifer) [inline src/app/packages/menu-items/layers-manager/reducers/layers.reducer.ts]: after a brief check in the internet I haven't found a better solution

File: src/app/packages/menu-items/layers-manager/services/data-layers.service.spec.ts (hunk 7 above)
Comment (MarkShleifer) [inline src/app/packages/menu-items/layers-manager/services/data-layers.service.spec.ts]: done

File: src/app/packages/menu-items/layers-manager/services/data-layers.service.ts (hunk 8 above)
Comment (MarkShleifer) [inline src/app/packages/menu-items/layers-manager/services/data-layers.service.ts]: done

File: src/app/effects/map.app.effects.spec.ts [hunk 9]
```diff
@@ -91,7 +90,14 @@ describe('MapAppEffects', () => {
 	})
 
 	it('addVectorLayer$ should add the selected Layer to the map', () => {
-		let staticLeaf: ILayerTreeNodeLeaf = { name: 'staticLayer', id: 'staticLayerId', isChecked: false, url: "fake_url", isIndeterminate: false, children: <ILayerTreeNode[]>[] };
+		let staticLeaf: ILayerTreeNodeLeaf = {
```
Comment (eldadcohen1) [inline src/app/effects/map.app.effects.spec.ts]: good

**Pull request merged**